<launch>

  <arg name="launch_prefix" default="" /> <!--set to value="gdbserver localhost:10000" for remote debugging-->
  <arg name="node_namespace" default="apriltag_ros_single_image_client_node" />
  <arg name="image_load_path" default="/home/fu/car/arm/photo/ApriltagRaw_1.png " /> <!-- Where to load image for analysis from -->
  <arg name="image_save_path" default="/home/fu/car/arm/photo/Apriltag_1.png" />  <!-- Where to save tag detections image -->

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg node_namespace)" />

  <node pkg="apriltag_ros" type="apriltag_ros_single_image_client_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >

    <param name="image_load_path" type="str" value="$(arg image_load_path)" />
    <param name="image_save_path" type="str" value="$(arg image_save_path)" />

    <!-- Camera intrinsic parameters -->
    <param name="fx" type="double" value="1097.2826519484" />
    <param name="fy" type="double" value="1093.696551235277" />
    <param name="cx" type="double" value="953.2646757356512" />
    <param name="cy" type="double" value="501.2671482091341" />

  </node>

</launch>
